code/game/objects/items/devices/radio/headset.dm

```diff
@@ -219,67 +219,139 @@
 	icon_state = "cargo_headset"
 	item_state = "headset"
 	frequency = PUB_FREQ
-	var/headset_hud_on = 1
+	var/datum/mob_hud/squadhud = null
+	var/mob/living/carbon/human/wearer = null
+	var/headset_hud_on = FALSE
+	var/sl_direction = FALSE
 	var/obj/machinery/camera/camera
 
-/obj/item/device/radio/headset/almayer/New()
+/obj/item/device/radio/headset/almayer/Initialize()
 	. = ..()
 	camera = new /obj/machinery/camera(src)
 	camera.network = list("LEADER")
 
 /obj/item/device/radio/headset/almayer/equipped(mob/living/carbon/human/user, slot)
-	if(slot == SLOT_EARS)
-		if(headset_hud_on)
-			var/datum/mob_hud/H = huds[MOB_HUD_SQUAD]
-			H.add_hud_to(user)
-			//squad leader locator is no longer invisible on our player HUD.
-			if(user.mind && user.assigned_squad && user.hud_used && user.hud_used.locate_leader)
-				user.hud_used.locate_leader.alpha = 255
-				user.hud_used.locate_leader.mouse_opacity = 1
+	if(slot == ITEM_SLOT_EARS)
```

Comment:
Do you understand the difference between `SLOT_EARS` and `ITEM_SLOT_EARS`?
One is a bitwise flag, the other is not.

Reply:
copy pasta when spooky's conflict resolution didn't compile. It's fixed